Waidhoferj/CSC-308-309-Project
# Added DiscoverArtworkMutation and made many edits surrounding metrics

To Test:
This request is quite tedious to test entirely because you would be manually checking if multiple metrics incremented by one for many different operations. This is not too necessary given that I have a test for the mutation and if anything, I made our metrics more reliable than they were before. With that said, you could experiment with mutations and see the user or artwork metrics update (ex: User creates artwork -> visit and create metrics updated; User discovers artwork -> visit metrics updated)

Questions:
Are the nonstop requests made to the backend new? When I boot up the backend, even without the frontend, it appears to make quite a bit of post requests (with some options requests). I am not sure why these are being done but this could've always been the case with the interface. 

NOTES: 
- This sets us up for more group functionality and makes the mutations more use-case specific
- Still need to add the "official" discover artwork functionality to frontend
- I made edits to a component on the front end in hopes that it will show the number of ratings for an artwork. My edits there didn't seem to do anything (or show the distance value as well).
